bluesky/bundlers.py

```diff
@@ -555,63 +555,74 @@ async def collect(self, msg):
 
         Expect message object is
 
-            Msg('collect', obj)
+            Msg('collect', collect_obj)
             Msg('collect', flyer_object, stream=True, return_payload=False)
         """
-        obj = msg.obj
+        collect_obj = msg.obj
 
         if not self.run_is_open:
             # sanity check -- 'kickoff' should catch this and make this
             # code path impossible
             raise IllegalMessageSequence(
                 "A 'collect' message was sent but no " "run is open."
             )
-        self._uncollected.discard(obj)
+        self._uncollected.discard(collect_obj)
 
-        if hasattr(obj, "collect_asset_docs"):
+        if hasattr(collect_obj, "collect_asset_docs"):
             # Resource and Datum documents
-            for name, doc in obj.collect_asset_docs():
+            for name, doc in collect_obj.collect_asset_docs():
                 # Add a 'run_start' field to the resource document on its way out.
                 if name == "resource":
                     doc["run_start"] = self._run_start_uid
                 await self.emit(DocumentNames(name), doc)
 
-        named_data_keys = obj.describe_collect()
-        # e.g., {name_for_desc1: data_keys_for_desc1,
-        #        name for_desc2: data_keys_for_desc2, ...}
+        collect_obj_configuration = {}
+        if hasattr(collect_obj, "read_configuration"):
+            doc_logger.debug("reading configuration from %s", collect_obj)
+            collect_obj_configuration = collect_obj.read_configuration()
+        else:
+            doc_logger.debug("%s has no read_configuration method", collect_obj)
+
         bulk_data = {}
-        local_descriptors = {}  # hashed on obj_read, not (name, objs_read)
-        for stream_name, data_keys in named_data_keys.items():
-            desc_key = stream_name
-            d_objs = frozenset(data_keys)
-            if desc_key not in self._descriptors:
+        local_descriptors = {}  # hashed on objs_read, not (name, objs_read)
+        # collect_obj.describe_collect() returns a dictionary like this:
+        #     {name_for_desc1: data_keys_for_desc1,
+        #      name_for_desc2: data_keys_for_desc2, ...}
+        for stream_name, stream_data_keys in collect_obj.describe_collect().items():
+            d_objs = frozenset(stream_data_keys)
+            if stream_name not in self._descriptors:
                 objs_read = d_objs
-                # We don't not have an Event Descriptor for this set.
+                # We do not have an Event Descriptor for this set.
                 descriptor_uid = new_uid()
-                object_keys = {obj.name: list(data_keys)}
+                stream_object_keys = {collect_obj.name: list(stream_data_keys)}
                 hints = {}
-                if hasattr(obj, "hints"):
-                    hints.update({obj.name: obj.hints})
+                # is this if block an error?
```

Comment:
The `hints` attribute is a relatively new thing, so we must treat it as optional to support old code. I think I would be in favor of keeping it optional; I don't think a minimal bluesky-compatible object should need to implement it.